### PR TITLE
fix: Broken CI

### PR DESCRIPTION
**Pull request by HarryET**
# Description
Fixes the broken/failing CI

## How Has This Been Tested?
Ran `cargo test` and `cargo clippy -- -D clippy::all` locally

## Due Dilligence

* [ ] Breaking change
* [ ] Requires a documentation update
* [ ] Requires a e2e/integration test update

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -30,7 +30,7 @@ jobs:
         os:
           - ubuntu-latest
         rust:
-          - stable
+          - nightly
```
Comment (heilhead) [inline .github/workflows/ci.yml:34]: Why nightly?

File: src/providers/noop.rs [hunk 2]
```diff
@@ -36,9 +36,9 @@ impl PushProvider for NoopProvider {
 impl NoopProvider {
     /// Insert empty notifications for a new token
     fn bootstrap(&mut self, token: String) {
-        if !self.notifications.contains_key(&token) {
-            self.notifications.insert(token, vec![]);
-        }
+        self.notifications
+            .try_insert(token, vec![])
```
Comment (heilhead) [inline src/providers/noop.rs]: Is this what the nightly for? And why change this code anyway?

Comment (HarryET) [reply src/providers/noop.rs]: Clippy dosen't like the if not exists, then insert

File: .github/workflows/ci.yml (hunk 1 above)
Comment (HarryET) [reply .github/workflows/ci.yml:34]: I thought we agreed on using nightly?

File: src/providers/noop.rs (hunk 2 above)
Comment (HarryET) [reply src/providers/noop.rs]: Fixed

File: .github/workflows/ci.yml (hunk 1 above)
Comment (HarryET) [reply .github/workflows/ci.yml:34]: Resolved in slack

**Review by heilhead — APPROVED** [review]
Great, thanks!

**Pull request merged**